### PR TITLE
Add execute_PayloadV1 rpc endpoint

### PR DESCRIPTION
**Pull request by avalonche**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
- Implements `engine_executePayloadV1` response as per Kintsugi specs

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ExecutePayloadV1Handler.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                executePayloadResult.EnumStatus = VerificationStatus.Syncing;
+                return ResultWrapper<ExecutePayloadV1Result>.Success(executePayloadResult);
+            }
+
+            ValidationResult result = ValidateRequestAndProcess(request, out Block? processedBlock);
+            if ((result & ValidationResult.AlreadyKnown) != 0 || result == ValidationResult.Invalid)
+            {
+                bool isValid = (result & ValidationResult.Valid) != 0;
+                executePayloadResult.EnumStatus = isValid ? VerificationStatus.Valid : VerificationStatus.Invalid;
+                executePayloadResult.LatestValidHash = isValid ? request.BlockHash : _blockTree.HeadHash;
```
Comment (avalonche) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/ExecutePayloadV1Handler.cs:92]: not sure if `_blockTree.HeadHash` retrieves the hash of the last valid block

File: src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutePayloadV1Result.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using System;
+using Nethermind.Core.Crypto;
+using Newtonsoft.Json;
+
+namespace Nethermind.Merge.Plugin.Data
+{
+    public class ExecutePayloadV1Result
+    {
+        [JsonProperty("status")]
+        public string? Status => Enum.GetName(typeof(VerificationStatus), EnumStatus)?.ToUpper();
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutePayloadV1Result.cs:27]: I added EngineStatus on themerge_kintsugi. Could we use it here too?

File: src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutePayloadV1Result.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+using System;
+using Nethermind.Core.Crypto;
+using Newtonsoft.Json;
+
+namespace Nethermind.Merge.Plugin.Data
+{
+    public class ExecutePayloadV1Result
+    {
+        [JsonProperty("status")]
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutePayloadV1Result.cs:26]: We can remove JsonProperty. It is not needed

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ExecutePayloadV1Handler.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/ExecutePayloadV1Handler.cs:92]: Yeah, good catch. I am not sure too. Let's ask specification authors.

**Pull request merged**